### PR TITLE
implement trio116, checking for trio.sleep(<duration over 24h>)

### LINKED ISSUE #71: Suggest `sleep_forever()` if `trio.sleep()` is passed a >24hr interval
```python
# These examples are probably not meant to ever wake up:
await trio.sleep(100000)
await trio.sleep(10**10)

# and these ones _definitely_ never wake up:
await trio.sleep(float("inf"))
await trio.sleep(math.inf)
await trio.sleep(1e999)  # 'inf literal' overflow trick
```

This is a pretty simple check, but using `await trio.sleep_forever()` instead does have a certain elegance - and the first example (real code) might actually wake up if you leave the process running for ~27 hours!

### PR DESCRIPTION
**Pull request by jakkdl**
fixes #71 

What felt like it should be a trivial check turned out to be a surprising amount of trouble:
* https://github.com/Zac-HD/hypothesmith/issues/27 completely broke tests, so I had to comment out all hypothesis functionality during testing. I'm guessing CI will break too?
* https://github.com/Zac-HD/flake8-trio/issues/73 wasn't too hard to figure out, but I've added some code to trio114 as well due to it. TODO: write tests for that (and/or split into separate PR)
* https://github.com/Zac-HD/shed/issues/60 if I wasn't a contributor to shed I'd just ignore why it was acting up, but was a tricky side-adventure to figure this out. And that's why files in `typings/` are modified despite me not using those.
* `ast` in python3.8 doesn't have `unparse`, so raising an error on `time.sleep(10**10)` would require a bunch of custom code. It could plausibly be worth it since that means we don't need an `eval`, but..... :shrug:
   * debugging it was fun though since I'm ignoring all exceptions in the eval call >.>;;
* I skipped TRIO115 ..... since i apparently mixed up the open issues expecting a new error relating to TRIO113/TRIO114.

Open questions:
1. Warn on duration < 0? Feels like it kind of never should happen, and will break on runtime, but is fairly trivial to add to this check - though would likely warrant a separate error code.
2. Should the code warn (i.e. just print something to stdout) if the eval fails? Any such code would break on runtime, and probably also caught by other linters, but completely ignoring exceptions .. can have consequences.
3. Aside from py38, 1272-1274 is entirely redundant since the eval can handle it. So there's a case for removing that code, or at least was until py38 decided to be a bother.
```python
            elif isinstance(arg, ast.Constant):
                if isinstance(arg.value, (int, float)) and arg.value > 86400:
                    self.error("TRIO116", node)
```

:sparkles:

### CODE REVIEW

File: flake8_trio.py [hunk 1]
```diff
@@ -1234,6 +1238,58 @@ def visit_AsyncFunctionDef(self, node: ast.AsyncFunctionDef):
...
+                for n in self.walk(arg)
+            ):  # pragma: no cover (since py38 can't do this)
+                try:
+                    # not strictly safe, and could take forever if passed
+                    # like 10**10**10**10
+                    if eval(ast.unparse(arg), None, None) > 86400:
+                        self.error("TRIO116", node)
+                except Exception:
+                    # warn? Should be caught by other linters or runtime though
+                    pass
```
[the remark this answers is not in the document]
Comment (jakkdl) [reply flake8_trio.py:1291]: Yeah, totes agree on skipping expressions, considered it myself as I was writing it.

**Comment by jakkdl** [comment]
Nothing to rebase, but fixed all comments and added tests for trio114. For trio113 I think it's actually impossible for missing visit's in those particular functions to cause problems, so no tests added (but visits are added to safeguard future headache).

**Comment by jakkdl** [comment]
Huh, we both missed that I'd written `# 22.11.5` with only a single leading `#` in the changelog, and not incremented `__version__`.

**Comment by Zac-HD** [comment]
> Huh, we both missed that I'd written `# 22.11.5` with only a single leading `#` in the changelog, and not incremented `__version__`.

If you'd done either alone, we have a test for consistency!  Worst-case here if I didn't catch it in more rigorous about-to-merge review is that we'd have not released this version, and probably noticed it was missing from PyPI 😁

**Review by Zac-HD — APPROVED** [review]
🚢 ✨

**Pull request merged**